state-recover/besu-plugin/src/main/kotlin/linea/staterecover/plugin/RecoveryModeManager.kt

```diff
@@ -0,0 +1,103 @@
+package linea.staterecover.plugin
+
+import linea.staterecover.RecoveryStatusPersistence
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+import org.hyperledger.besu.plugin.data.AddedBlockContext
+import org.hyperledger.besu.plugin.services.BesuEvents
+import org.hyperledger.besu.plugin.services.mining.MiningService
+import org.hyperledger.besu.plugin.services.p2p.P2PService
+import org.hyperledger.besu.plugin.services.sync.SynchronizationService
+import java.util.concurrent.atomic.AtomicBoolean
+
+class RecoveryModeManager(
+  private val synchronizationService: SynchronizationService,
+  private val p2pService: P2PService,
+  private val miningService: MiningService,
+  private val recoveryStatePersistence: RecoveryStatusPersistence
+) :
+  BesuEvents.BlockAddedListener {
+  private val log: Logger = LogManager.getLogger(RecoveryModeManager::class.java.name)
+  private val triggered = AtomicBoolean(false)
+  private var currentBlockNumber: ULong = 0u
+  val targetBlockNumber: ULong?
+    get() = recoveryStatePersistence.getRecoveryStartBlockNumber()
+
+  val headBlockNumber: ULong
+    get() = currentBlockNumber
+
+  // Enum representing the states of recovery mode
+  private enum class RecoveryModeState {
+    NORMAL_MODE,
+    RECOVERY_MODE
+  }
+
+  private var recoveryModeState = RecoveryModeState.NORMAL_MODE
```

Comment:
I couldn't find any meaningful usage of it, other than switching the mode. What is the difference between recovery mode and normal mode?